docs/extensibility/extending-blocks.md

```diff
@@ -38,6 +38,43 @@ wp.hooks.addFilter(
 
 A filter that applies to the result of a block's `save` function. This filter is used to replace or extend the element, for example using `wp.element.cloneElement` to modify the element's props or replace its children, or returning an entirely new element.
 
+Notably, while you _can_ modify a block's `save` result from a theme or plugin, this behavior won't persist when the theme or plugin is deactivated.
+
+_Example:_
+
+Include a custom data attribute on the Image Block `<img>` tag when the block attribute is set.
+
+```js
+function addCustomDataAttribute( element, blockType, attributes ) {
+	if ( 'core/image' !== blockType.name ) {
+		return element;
+	}
+
+	if ( ! attributes.myCustomAttribute.length ) {
+		return element;
+	}
+	// Modifying nested JavaScript objects is pretty awful.
+	element = lodash.assign( {}, element, {
+		props: lodash.assign( {}, element.props, {
+			children: [
+				lodash.assign( {}, element.props.children[0], {
```

Comment:
This will not assign `data-custom-attribute` when the image is assigned a URL. The deep override should be a signal of its fragility.

Reply:
I don't follow this either. Can you explain further?

Reply:
The markup shape changes depending on whether the image block has a `url` attribute assigned.

Aside: It's maybe telling that this is an easy thing to overlook.